processing/src/main/java/io/druid/query/aggregation/AggregatorFactory.java

```diff
@@ -60,14 +60,30 @@
    *
    * @return a new Factory that can be used for operations on top of data output from the current factory.
    */
-  public AggregatorFactory getCombiningFactory();
+  public abstract AggregatorFactory getCombiningFactory();
+
+  /**
+   * Returns an AggregatorFactory that can be used to merge the output of aggregators from this factory and
+   * other factory.
+   * This method is relevant only for AggregatorFactory which can be used at ingestion time.
+   *
+   * @return a new Factory that can be used for merging the output of aggregators from this factory and other.
+   */
+  public AggregatorFactory getMergingFactory(AggregatorFactory other) throws AggregatorFactoryNotMergeableException
```

Comment:
merging of aggregator from several input segments would return null if merging was not possible for any reason.
this can happen via AppendTask and resulting segment would not have aggregators in its metadata. Optionally, user can explicitly supply the list of aggregators to be used and merge logic wouldn't be used.
In other situations aggregators are explicit (e.g. MergeTask , RealtimeIndexTask, HadoopIndexTAsk etc etc) so merging logic is not used.
In a future PR for coordinator merging via hadoop index task, it can probably choose to fail in that situation and user can manually run the hadoop re-indexing to rectify the situation.